Alicerce/Sources/Logging/Log.swift

```diff
@@ -0,0 +1,124 @@
+//
+//  Alicerce.swift
+//  Alicerce
+//
+//  Created by Meik Schutz on 03/04/17.
+//  Copyright © 2017 Mindera. All rights reserved.
+//
+
+import Foundation
+
+public final class Log {
+
+    public private(set) static var destinations = [LogDestination]()
+
+    public static let defaultLevel = Level.error
+
+    // MARK:- Destination Management
+
+    public class func register(_ destination: LogDestination) {
+        if destinations.contains(where: { $0.instanceId == destination.instanceId }) == false {
+            destinations.append(destination)
+        }
+    }
+
+    public class func unregister(_ destination: LogDestination) {
+        destinations = destinations.filter({ registeredDestinaton -> Bool in
+            return registeredDestinaton.instanceId != destination.instanceId
+        })
+    }
+
+    public class func removeAllDestinations() {
+        destinations.removeAll()
+    }
+
+    // MARK:- Logging
+
+    public class func verbose(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .verbose, message: message, file: file, function: function, line: line)
+    }
+
+    public class func debug(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .debug, message: message, file: file, function: function, line: line)
+    }
+
+    public class func info(_ message: @autoclosure () -> String,
+                           file: String = #file,
+                           function: String = #function,
+                           line: Int = #line) {
+
+        log(level: .info, message: message, file: file, function: function, line: line)
+    }
+
+    public class func warning(_ message: @autoclosure () -> String,
+                              file: String = #file,
+                              function: String = #function,
+                              line: Int = #line) {
+
+        log(level: .warning, message: message, file: file, function: function, line: line)
+    }
+
+    public class func error(_ message: @autoclosure () -> String,
+                            file: String = #file,
+                            function: String = #function,
+                            line: Int = #line) {
+
+        log(level: .error, message: message, file: file, function: function, line: line)
+    }
+
+    public class func log(level: Level,
+                          message: @autoclosure () -> String,
+                          file: String = #file,
+                          function: String = #function,
+                          line: Int = #line) {
+
+        let item = Log.Item(level: level, message: message(), file: file,
```

Comment:
I think we should have a dispatch queue to handle logging.

If we want to open this to the community in order to create theirs `Destination` classes, we have to make sure that none of them can potentially lock the caller queue, even if is a background thread.
This should be logger responsibility.
This is just my idea :)